R/ggproto.r

```diff
@@ -74,8 +82,17 @@ fetch_ggproto <- function(x, name) {
   } else {
     # If not found here, recurse into super environments
     super <- .subset2(x, "super")
-    if (is.ggproto(super))
-      res <- fetch_ggproto(super, name)
+    if (is.null(super)) {
+      # no super class
```

Comment:
@vivekktiwari If there is no super class, do we assign a default class here?

Reply:
@faizan-khan-iit If it is not inheriting any superclass, I think NULL goes as Default automatically as some functions may not have any parent class to catch.

Reply:
Then please add a statement explicitly returning `NULL` here along with a comment. But please make sure that is the case before making the changes..

Reply:
Please look at the ```res``` variable, I think it is doing the same work, since in ```if``` statement, it does nothing, ```res``` remains ```NULL``` and ignores all other if and else statements and finally at the end it returns ```res``` as ```NULL```

**Please look at the second last line of the code**

```R
fetch_ggproto <- function(x, name) {
  res <- NULL

  val <- .subset2(x, name)
  # The is.null check is an optimization for a common case; exists() also
  # catches the case where the value exists but has a NULL value.
  if (!is.null(val) || exists(name, envir = x, inherits = FALSE)) {
    res <- val
  } else {
    # If not found here, recurse into super environments
    super <- .subset2(x, "super")
    if (is.null(super)) {
      # no super class
    } else if (is.function(super)) {
      res <- fetch_ggproto(super(), name)
    } else {
      stop(
        class(x)[[1]], " was built with an incompatible version of ggproto.\n",
        "Please reinstall the package that provides this extension.",
        call. = FALSE
      )
    }
  }

  res 
#In the end it is return res here, and at the start of the function, **res <- NULL**
}